tensorflow_model_optimization/python/core/sparsity_tf2/riglpruner_test.py

```diff
@@ -0,0 +1,577 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for the key functions in riglpruner library."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+# import g3
+
+from absl.testing import parameterized
+import numpy as np
+import tensorflow as tf
+
+# TODO(b/139939526): move to public API.
+from tensorflow.python.keras import keras_parameterized
+from tensorflow_model_optimization.python.core.sparsity_tf2 import schedule
+from tensorflow_model_optimization.python.core.sparsity_tf2 import riglpruner as pruner
+from tensorflow_model_optimization.python.core.sparsity_tf2 import pruning_optimizer
+from tensorflow_model_optimization.python.core.sparsity_tf2 import pruning_config
+
+dtypes = tf.dtypes
+test = tf.test
+
+
+def make_update_schedule(fraction, begin, end, freq):
+  return schedule.ConstantSparsity(fraction, begin, end, freq)
+
+def sample_noise(x, mu=0, sigma=1., dtype=tf.float64):
+  sample = tf.random.normal((), mean=mu,  stddev=sigma, dtype=dtype)
+  return sample
+
+def _dummy_gradient(x, n=100, dtype=tf.float64):
+  try:
+    base_type = x.dtype
+  except:
+    base_type = dtype
+  # must shuffle otherwise each gradient update will be the same and 
+  # connections are never regrown
+  grad = tf.random.shuffle(np.linspace(1., 100., n))
```

Comment:
Using a random shuffle would invalidate the determinism of the gradients, no?

Might be better to take `step` as an input & predictably generate a dummy gradient for each one.